06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js

```diff
@@ -0,0 +1,178 @@
+const alice = require('../books/alice.json');
+const rabbits = require('../books/rabbits.json');
+const wrongkeys = require('../books/wrongkeys.json');
+const oneKey = require('../books/oneKey.json');
+const emptyString = require('../books/emptystring.json');
+
+describe('InvertedIndex Test Suite', () => {
+  beforeAll(() => {
+    this.invertedIndex = new InvertedIndex();
+    this.invertedIndex.createIndex('alice', alice);
+    this.invertedIndex.createIndex('rabbits', rabbits);
+    const eventListener = jasmine.createSpy();
+    const dummyFileReader = { addEventListener: eventListener };
+    spyOn(window, 'FileReader').and.returnValue(dummyFileReader);
+  });
+
+  describe('InvertedIndex Class', () => {
+    it('should be defined', () => {
+      expect(InvertedIndex).toBeDefined();
+    });
+
+    it('should create inverted index instance', () => {
+      expect(this.invertedIndex instanceof InvertedIndex).toBe(true);
+    });
+
+    it('should have object as the instance of invertedIndex', () => {
+      expect(typeof this.invertedIndex).toEqual('object');
+    });
+
+    it('should have an indexed object to hold all indexes', () => {
+      expect(typeof this.invertedIndex.indexed).toEqual('object');
+    });
+  });
+
+  describe('The readFile method', () => {
+    it('should read a file', () => {
+      const reader = new FileReader();
+      reader.addEventListener('load', (e) => {
+        const fileRead = reader.result;
+        expect(fileRead).toBeTruthy();
+        expect(JSON.parse(fileRead) instanceof Object).toBeTruthy();
+        done();
+      });
+    });
+
+    it('should call validateFile', () => {
+      const reader = new FileReader();
+      reader.addEventListener('load', () => {
+        const fileRead = JSON.parse(reader.result);
+        expect(InvertedIndex.validateFile(fileRead)).toHaveBeenCalled;
+        done();
+      });
+    });
```

Comment:
I am not completely sure about this test spec. Does it not belong in a different test suite?